### PR TITLE
Master2

### PR DESCRIPTION
**Pull request by Bossmanswift**
This is a pull request that highlights changes and additions to the project.

I created a WarehouseTracker class, FreightType Enum, and made slight modifications to the Shipment and warehouse classes. I also created a JUnit Test class and some test cases. Please review this and give feedback to me. I'll be glad to answer any questions you have.

### CODE REVIEW

**Review by jtohsohr — APPROVED** [review]

File: src/com/generic/tracker/WarehouseTracker.java [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+ * 
+ * THE MAJOR REASON I MAKE THE METHODS RETURN SOMETHING IS FOR TESTING
+ * THE ONLY METHOD THAT SHOLD BE ALLOWED TO HAVE 
+ * A VOID RETURN TYPE IN THIS CLASS IS printWarehouseDetails() and addWarehouse
+ * 
+ * @author Seyi Ola
+ *
+ */
+public class WarehouseTracker {
+
```
Comment (konglor) [inline src/com/generic/tracker/WarehouseTracker.java:26]: I like the addition of this class

File: src/com/generic/model/FreightType.java [hunk 2]
```diff
@@ -0,0 +1,3 @@
+package com.generic.model;
+
+public enum FreightType {air, truck, ship, rail}
```
Comment (konglor) [inline src/com/generic/model/FreightType.java:3]: enum entries should be capitalized. AIR, TRUCK, SHIP, RAIL

File: src/com/generic/tracker/WarehouseTracker.java [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+	 * and if the shipment was successfully added, it 
+	 * returns 1.
+	 * @param warehouseID the warehouseID
+	 * @param shipment the shipment to add to this warehouse
+	 * @return -1 if ID does not exist
+	 * 		  , 0 if freight receipt is disabled,
+	 * 		  , 1 if successfully added.
+	 * 			
+	 */
+	public int addShipment(int warehouseID, Shipment mShipment)
```
Comment (konglor) [inline src/com/generic/tracker/WarehouseTracker.java:69]: If you are going to return an int with a possible result of 3 values,
return an ENUM for self-code documentation.

I would prefer just a boolean value for simplicity.

File: src/com/generic/tracker/WarehouseTracker.java [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+	 * and if the warehouse freight is already enabled, it 
+	 * returns 1.
+	 * @param warehouseID the warehouse ID
+	 * @return log number based on result(-1:non-existent warehouse,
+	 * 									   0: already enabled,
+	 * 									   1: successfully added).
+	 */
+	
+	 public int enableFreight(int warehouseID)
+	 {
```
Comment (konglor) [inline src/com/generic/tracker/WarehouseTracker.java:100]: same here

File: src/com/generic/tracker/WarehouseTracker.java [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+	 * is disabled already, it returns 0, and if the warehouse freight is already
+	 * disabled, it returns 1.
+	 * @param warehouseID the warehouse ID
+	 * @return log number based on result(-1:non-existent warehouse,
+	 * 									   0: already disabled,
+	 * 									   1: successfully added).
+	 */
+
+	public int endFreight(int warehouseID) {
+		Warehouse theWarehouse = warehouses.get(warehouseID);
```
Comment (konglor) [inline src/com/generic/tracker/WarehouseTracker.java:129]: If a warehouse does not exist, it should throw a RunTimeException.
the other can be boolean

File: .classpath [hunk 6]
```diff
@@ -1,11 +1,13 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <classpath>
 	<classpathentry kind="src" path="src"/>
+	<classpathentry kind="src" path="QATester"/>
```
Comment (konglor) [inline .classpath:4]: this is just by convention, but path should be "unit"

File: QATester/com/generic/QATester/QATester.java [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+	 * mapped to object's instance. This implies that
+	 * the warehouse key will always be same as the
+	 * warehouseID field
+	 * @param warehouseID pre-defined(already existing) set of warehouseIDs
+	 */
+	@ParameterizedTest
+	@ValueSource(ints = {12513,15566})
+	void testWarehouseTrackerIDKeyMapping(int warehouseID) 
+	{
+		assertEquals(warehouseID, wTracker.getWarehouse(warehouseID).getWarehouseID());
```
Comment (konglor) [inline QATester/com/generic/QATester/QATester.java:79]: Tests should be written in the AAA format-
Arrange, Act, Assert.

File: QATester/com/generic/QATester/QATester.java [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+
+
+/**
+ * A Junit class for unit testing
+ * @author Seyi Ola
+ *
+ */
+
+public class QATester {
+
```
Comment (konglor) [inline QATester/com/generic/QATester/QATester.java:23]: For example, if you want to test Warehouse class,
by convention, you create a file called WarehouseTest.java
and write it's test inside of said file.

and if you want to test Shipment class, you would create a file called ShipmentTest.

This creates modular tests that can be executed as a standalone

**Review by konglor — CHANGES_REQUESTED** [review]
slightly off conventions but good.

File: src/com/generic/tracker/WarehouseTracker.java (hunk 1 above)
Comment (Bossmanswift) [reply src/com/generic/tracker/WarehouseTracker.java:26]: Thanks

File: src/com/generic/model/FreightType.java (hunk 2 above)
Comment (Bossmanswift) [reply src/com/generic/model/FreightType.java:3]: Fixed

File: src/com/generic/tracker/WarehouseTracker.java (hunk 3 above)
Comment (Bossmanswift) [reply src/com/generic/tracker/WarehouseTracker.java:69]: Returns boolean now

File: src/com/generic/tracker/WarehouseTracker.java (hunk 4 above)
Comment (Bossmanswift) [reply src/com/generic/tracker/WarehouseTracker.java:100]: Returns boolean now

File: src/com/generic/tracker/WarehouseTracker.java (hunk 5 above)
Comment (Bossmanswift) [reply src/com/generic/tracker/WarehouseTracker.java:129]: Returns boolean for now

File: QATester/com/generic/QATester/QATester.java (hunk 7 above)
Comment (Bossmanswift) [reply QATester/com/generic/QATester/QATester.java:79]: Noted

File: QATester/com/generic/QATester/QATester.java (hunk 8 above)
Comment (Bossmanswift) [reply QATester/com/generic/QATester/QATester.java:23]: Noted

**Pull request closed without merging**